tests/LogButton.test.js

```diff
@@ -1,16 +1,44 @@
 import React from 'react';
+import { Provider } from 'react-redux';
 import renderer from 'react-test-renderer';
+import configureStore from 'redux-mock-store';
 
-import { LogButton } from '../src/components/LogButton';
+const mockStore = configureStore([]);
 
+import LogButton from '../src/components/LogButton';
 
 describe('<LogButton />', () => {
+    let store;
+    var toLocaleString = Date.prototype.toLocaleString;
+
+    beforeEach(() => {
+        store = mockStore({
+            count: {
+                habits: {
+                    1: {
+                        id: 1,
+                        title: 'My First Habit',
+                        count: 1,
+                        limit: 1,
+                        log: [{ updated: 1612235045000, info: { type: 'created' } }, { updated: 1612235045000, info: { type: 'increment' } }]
+                    }
+                }
+            }
+        });
+        Date.prototype.toLocaleString = function () {
+            return '02/02/2021, 03:04:05';
+        }
+    });
+    afterAll(() => {
+        Date.prototype.toLocaleString = toLocaleString;
+    });
```

Comment:
Can I ask why we need to mock out the `toLocaleString` method here?

Reply:
Before each test I'm mocking the method so it returns a particular string (to get around the failing CI issues related to different locales), in line 33, just setting it back to as was before my mock. Is there a better way to do this? I couldn't find alternative way to get around the locale issue.

Reply:
It looks like `toLocaleString` takes a locale as an argument: `myDate.toLocaleString('en-GB')`

It might be that we have `en-GB` as our default locale but CI is in the US so has `en-US`?

Reply:
The toLocaleString is part of the code we want to test :
 Line 76  of LogButton component:  updatedDisplay: event ? (new Date(event.updated)).toLocaleString() : 'Never',
- Can't pass an en-US locale in the actual code as not all users will be in that region - hence the mocking. 
- I wouldn't have thought we could pass en-US in the mock either as then my local tests would fail?
- Maybe this is something we can pick up in our next session